src/maps/mod.rs

```diff
@@ -464,15 +469,59 @@ impl PerfMap {
                     let sample = PerfEventSample { header, size, data };
                     Ok(Some(PerfEvent::Sample(sample)))
                 }
-                perf_event_type::PERF_RECORD_LOST => Ok(Some(PerfEvent::<'a>::Lost(
-                    &*(buf.as_ptr() as *const PerfEventLostSamples),
-                ))),
+                perf_event_type::PERF_RECORD_LOST => {
+                    let lost = &*(buf.as_ptr() as *const PerfEventLostSamples);
+                    Ok(Some(PerfEvent::Lost(lost.count)))
+                }
                 _ => Ok(None),
             }
         }
     }
+
+    /// Reads all available events
+    ///
+    /// Stops reading either on a real error (and propagates it) or on
+    /// a OxidebpfError::NoPerfData and returns `None`.
+    pub(crate) fn read_all(&self) -> impl Iterator<Item = Result<PerfEvent, OxidebpfError>> + '_ {
```

Comment:
this could potentially be "smarter" by not having to redo some of the setup in `read` every time but this is a good first step I think. `libbpf` does have basically an "event emitter" when it comes to read ebpf events so I think having an iterator is the rusty way of doing that.